src/ui/Startup.cs

```diff
@@ -194,12 +202,14 @@ public void ConfigureServices(IServiceCollection services)
             });
 
             services.AddSingleton<IFeatureFlags>(x => new FeatureFlags(Configuration.GetSection("features")));
+            services.AddSingleton<ISearchAndCompareUrlService>(x => new SearchAndCompareUrlService(Configuration.GetValue("SearchAndCompare:UiBaseUrl", "")));
             services.AddSingleton<IManageCoursesApiClientConfiguration, ManageCoursesApiClientConfiguration>();
             services.AddScoped(provider => AnalyticsPolicy.FromEnv());
             services.AddScoped<AnalyticsAttribute>();
             services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
             services.AddSingleton<ManageCoursesConfig, ManageCoursesConfig>();
             services.AddSingleton<IManageApi, ManageApi>();
+            services.AddScoped<ICourseDetailsService>(provider => new CourseDetailsService(provider.GetService<IManageApi>(), new CourseMapper()));
```

Comment:
I think we can probably get the params of this to be DI'd so you don't need the hard-coded new. Perhaps with the following:

```
services.AddScoped<CourseMapper,CourseMapper>;
services.AddScoped<ICourseDetailsService,CourseDetailsService>;
```

minor